app/models/edition.rb

```diff
@@ -103,6 +103,15 @@ class Edition < ApplicationRecord
     joins(revision: :metadata_revision).where(sql, political: political)
   end
 
+  scope :history_mode, ->(history_mode = true) do
+    if history_mode
+      political.where.not(government_id: [nil, Government.current.id])
+    else
+      political(false)
+        .or(political.where(government_id: [nil, Government.current.id]))
+    end
```

Comment:
I've been thinking about this one - I wonder if there's any way we can reuse the logic on line 108 in line 111. Something along the lines of "the inverse of the above". But I suspect there isn't a way of doing that that retains the database querying efficiency of what you've written 🤔 

I'm also wondering about the repetition of this business logic in two places - in the database query, and in the application method `history_mode?` you added in the previous commit. I suspect there's little we can do about this, but there's definitely a risk our business logic could go out of sync in the two places if we're not careful.

Reply:
Sadly to the best of my knowledge there isn't a nice way we can't do an inverse of a Rails scope. I'm not really sure why not as it's probably just the where condition wrapped in a not, but I guess scopes can do a lot more than just where conditions. I've googled this a few times and never found anything particularly nice.

You're right it is a risk about the logic duplication, I'm hoping it is reduced by things being quite atomic (for example reusing political scope). I'm not really sure there's a way to avoid it though. One option would be only relying on db queries, but that seems a bit superfluous and wouldn't work on anything not persisted. I'm ok with this as it is but agree it's a risk.